common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java

```diff
@@ -447,6 +453,53 @@ private boolean isAmazonAppRedirect(@NonNull final String url) {
         return url.startsWith(AMAZON_APP_REDIRECT_PREFIX);
     }
 
+    private boolean isWebCpEnrollmentUrl(@NonNull final String url) {
+        return url.startsWith(AuthenticationConstants.Broker.WEBCP_ENROLLMENT_URL);
+    }
+
+    private boolean isWebCpAuthorizeUrl(@NonNull final String url) {
+        // URL should be for authorize request for webcp.
+        final String methodTag = TAG + ":isWebCpAuthorizeUrl";
+        try {
+            final URI uri = new URI(url);
+            final String host = uri.getHost();
+            final String path = uri.getPath();
+
+            if (host == null || path == null) {
+                Logger.info(methodTag, "URL missing host or path");
+                return false;
+            }
+
+            if (!AzureActiveDirectory.isValidCloudHost(new URL(url))) {
+                Logger.info(methodTag, "URL host is not a valid Azure cloud host");
+                return false;
+            }
+
+            if (!path.contains("/authorize")) {
+                Logger.info(methodTag, "URL path does not contain /authorize");
+                return false;
+            }
+
+            final Map<String, String> queryParams = StringExtensions.getUrlParameters(url);
+            final String clientId = queryParams.get(AuthenticationConstants.OAuth2.CLIENT_ID);
+
+            if (StringUtil.isNullOrEmpty(clientId)) {
+                Logger.info(methodTag, "Authorize URL does not contain client_id");
+                return false;
+            }
+
+            final boolean isWebCpClient = AuthenticationConstants.Broker.WEBCP_CLIENT_ID.equalsIgnoreCase(clientId);
+            Logger.info(methodTag, isWebCpClient
+                    ? "WebCP authorize URL contains valid WebCP client_id."
+                    : "Authorize URL contains different client_id.");
```

Comment:
> "Authorize URL contains different client_id."

This message can be a bit misleading. May be change to `Not running WebCP flow as client_id in authorize is not webcp client_id`

Reply:
Will fix this in next PR I am raising for flighting with tenants. Builds on this PR have passed and do not want to make any changes since this is a minor one,